src/control/ros/Conversions.cpp

```diff
@@ -173,8 +282,29 @@ void extractTrajectoryPoint(
 
   Eigen::VectorXd tangentVector;
   auto state = space->createState();
+
+  auto prevState = space->createState();
+  space->convertPositionsToState(previousPoint, prevState);
+
   trajectory->evaluate(timeAbsolute, state);
-  space->logMap(state, tangentVector);
+  space->convertStateToPositions(state, tangentVector);
+
+  auto geodesicInterpolator
+      = std::dynamic_pointer_cast<aikido::statespace::GeodesicInterpolator>(
+          interpolator);
+  if (!geodesicInterpolator)
+  {
+    throw std::invalid_argument(
+        "The interpolator of trajectory should be a GeodesicInterpolator");
+  }
+
+  auto diff = geodesicInterpolator->getTangentVector(prevState, state);
```

Comment:
I'm not entirely sure I understand why this change is necessary. So far, my understanding of this PR is:
- We plan with SO(2) joints
- We convert SO(2) joints to R1, postprocess (because postprocessors only deal with Rn joints), then convert back to SO(2)

What are we trying to do in this function? Is it what I thought was happening in https://github.com/personalrobotics/aikido/pull/496#discussion_r253279846, i.e. basically converting this into an Rn-joint trajectory to send out via ROS?

My main qualm is that it feels wrong to be requiring a specific type of interpolator here. I'm wondering whether we can rewrite this without using `GeodesicInterpolator`-specific methods, perhaps by actually converting it to an Rn trajectory and then just using `logMap`?

Reply:
To my understanding, we are committing to executing the shortest paths i.e. interpolating via the Geodesic anytime we execute a trajectory.

> What are we trying to do in this function?
We want to extract waypoints along the trajectory "ensuring that the interpolation happens along the geodesic so that SO2 joint space is handled appropriately". 

What is the PR doing?
- We plan with SO2 joints
- We convert SO2 joints to R1 and time it -> this generates a spline that has been timed appropriately.
- When converting to ROS trajectory, each waypoint from this spline is extracted using the trajectory's statespace which is SO2 and hence requiring us to use the previous point to be able to interpolate appropriately, and then use the converted extracted waypoint to construct the ROS trajectory.

Conversion into Rn-joint trajectory and then using logMap would also involve the geodesic interpolator so I am not sure how that would help.

1. The issue is that the base Interpolator class does not have a `getTangentVector()` and I don't see why it shouldn't since every derived class would need to support that. So I can alternatively add a virtual function there and if we take the interpolator by reference or obtain it from the trajectory, it should point to the right implementation? 
2. But the problem is we need to be able to support both Interpolated trajectories and Splines. Spline trajectories do not have an interpolator attached to them so we need to use _something_.

Reply:
@aditya-vk I didn't fully get the 3rd bullet po int. `trajectory` is already  in `MetaSkeletonStateSpace` so it's a Cartesian product of SO2. In SO2 spline trajectory, why do we need the previous point to interpolate appropriately?

Reply:
Yes exactly. It is in the Cartesian Product of SO2, call it `space`. So when we evaluate the trajectory, `space` is used to retrieve the `state`. Although the spline polynomial is correct, when we retrieve it using `space`, the representation for SO2 joints is converted back to between `-pi` and `pi`.

Reply:
This leaves two options: Either have your trajectory be tied with `CartesianProduct` space of all `R1` joints which can again get messy since we check for `MetaSkeletonStateSpace` during execution or use the interpolation from the previous point.

Reply:
I was wondering if we could return `evaluatePolynomial()` instead of `evaluate()` since that returns the value of the polynomial which is what we desire. However, again, since the spline segments are solved from 0, we need to know the start state of the segment which is stores in `mSegments.startState` which is represented as R + SO2. So this route is pretty much useless as well.